R/assertions.R

```diff
@@ -0,0 +1,82 @@
+#' @export
+validate <- function(data, name, description = NULL) {
+  if (missing(name)) {
+    name <- deparse(substitute(data))
+    if (name == ".") {
+      name <- get_first_name(data)
+    }
+  }
+  attr(data, "data-description") <- description
+  attr(data, "data-name") <- name
+  attr(data, "assertr_in_chain_success_fun_override") <- assertr::success_append
+  attr(data, "assertr_in_chain_error_fun_override") <- assertr::error_append
+  data
+}
+
+#' @export
+assert_if <- function(data, expr, description = NA, obligatory = FALSE) {
+  assertr::verify(data = data,
+                  expr = !!rlang::enexpr(expr),
+                  description = format_description(description),
+                  skip_chain_opts = FALSE,
+                  obligatory = obligatory,
+                  success_fun = assertr::success_append,
+                  error_fun = assertr::error_append,
+                  defect_fun = assertr::defect_append)
+}
+
+#' @export
+assert_cols <- function(data, predicate, ..., predicate_calculated_from_column = FALSE, obligatory = FALSE, description = NA) {
+  if (predicate_calculated_from_column) {
+    data <- assertr::insist(
+      data = data, predicate_generator = predicate, ...,
+      skip_chain_opts = FALSE,
+      obligatory = obligatory,
+      description = format_description(description),
+      success_fun = assertr::success_append,
```

Comment:
@przytu1 I'd prefer the approach initially proposed by me: https://github.com/Appsilon/data.validator/pull/8/commits/4f90f83c9d3c3f5e8f5bb7f4ea06e301f08a1a80
This implementation limits the features offered by assertr a lot. Any parameter like `skip_chain_opts` cannot be overwritten in the external one (parameters duplication error).

Reply:
Good point. I agree.